### PR TITLE
improve: remove EventSourceInitializer

### PR DESCRIPTION
**Pull request by csviri**
Move `EventSourceIntiailizer` directly to `Reconciler`. Based on the patter, if properly done, there will be always EventSource. Still can return empty map.

### CODE REVIEW

**Comment by csviri** [comment]
We should consider (I'm leaning to this) not even provide a default implementation, although might be confusing for someone who starts first, that is that part not simple to understand first (like how Informers works and why essential), but at least moves users news to operators to good direction.

what do you tink @metacosm @jcechace  ?

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Reconciler.java [hunk 1]
```diff
@@ -14,6 +20,78 @@ public interface Reconciler<R extends HasMetadata> {
...
+  }
+
+  /**
+   * Utility method to easily create map with generated name for event sources. This is for the use
+   * case when the event sources are not access explicitly by name in the reconciler.
+   *
+   * @param eventSources to name
+   * @return even source with default names
+   */
+  static Map<String, EventSource> nameEventSources(EventSource... eventSources) {
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Reconciler.java:45]: I would prefer that we move all these methods to a different utilty class. They have nothing to do with Reconciler itself and pollute its API, imo.

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/EventSource.java [hunk 2]
```diff
@@ -8,9 +8,7 @@
 /**
  * Creates an event source to trigger your reconciler whenever something happens to a secondary or
  * external resource that would not normally trigger your reconciler (as the primary resources are
- * not changed). To register EventSources with so that your reconciler is triggered, please make
- * your reconciler implement
- * {@link io.javaoperatorsdk.operator.api.reconciler.EventSourceInitializer}.
+ * not changed). To register EventSources with so that your reconciler is triggered.
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/EventSource.java:11]: This doesn't make sense anymore.

**Comment by csviri** [comment]
> The default implementation should stay because, imo, most people will never need to implement this method. What would the implementation be for someone who only uses managed workflows, for example?

ok that is good argument, to keep the default impl.

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Reconciler.java (hunk 1 above)
Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Reconciler.java:45]: Will move it for now, was thinking to also have have EventSources "Named" like now with the Dependent Resources. Will make a separate PR. That would make these methods basically obsolete.

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/EventSource.java (hunk 2 above)
Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/EventSource.java]: changed.

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Reconciler.java (hunk 1 above)
Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Reconciler.java]: Added `EventSourceUtils`.

**Review by metacosm — CHANGES_REQUESTED** [review]
The documentation needs updates.

File: docs/documentation/dependent-resources.md [hunk 3]
```diff
@@ -307,16 +306,12 @@ public class WebPageStandaloneDependentsReconciler
 There are multiple things happening here:
 
 1. Dependent resources are explicitly created and can be access later by reference.
-2. Event sources are produced by the dependent resources, but needs to be explicitly registered in
-   this case by implementing
-   the [`EventSourceInitializer`](https://github.com/java-operator-sdk/java-operator-sdk/blob/main/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/EventSourceInitializer.java)
-   interface.
-3. The input html is validated, and error message is set in case it is invalid.
-4. Reconciliation of dependent resources is called explicitly, but here the workflow
+2. The input html is validated, and error message is set in case it is invalid.
```
[the remark this answers is not in the document]
Comment (csviri) [reply docs/documentation/dependent-resources.md:309]: I think now this is fixed by you :)

[the remark this answers is not in the document]
Comment (metacosm) [reply docs/documentation/dependent-resources.md:309]: No, the example and its documentation still needs to be updated.

**Review by metacosm — APPROVED** [review]

**Review by metacosm — APPROVED** [review]

**Pull request merged**